_posts/2020-05-03-static-ipv6-playground.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Static IPv6 Playground"
+date: 2020-05-03
+categories:
+  - blog
+tags:
+  - ipv6
+---
+
+This post is all about setting up an IPv6 enviorment for learning and testing 
+of varios setups and scenarios.
+
+We will use network namespaces for our setup and proceed to define static
+IP addresses.
+(we will cover dynamic IP addresses in a following post)
+
+## Basic Setup
+In order to prepare the ground for our IPv6 playground, we will use network
+namespaces to simulate different network stacks.
+A network namespace is an isolated network stack that includes interfaces,
+ip addresses and routes.
+
+Throught this post, we will use the `ip` command which is part of the
+[iproute2](https://wiki.linuxfoundation.org/networking/iproute2) utilities.
+
+### Namespace creation
+For our setup, we will use two namespaces: red & blue.
+
+Lets create our two namespaces:
+```
+sudo ip netns add red
+sudo ip netns add blue
+```
+Each namespace is created with a loopback interface which requires an explicit
+enablement:
+```
+sudo ip netns exec red ip link set lo up
+sudo ip netns exec blue ip link set lo up
+```
+
+### L2 connectivity (the veth)
+In order to enable connectivity between the two namespaces, we will use a
+veth interface. A veth interface type comes always in pairs, anything that
+ingress one edge, egress the other edge and vice versa. It provides a L2 local
+connectivity between the peers.
+
+We will create the veth interface at the root namespace and then place each
+peer in one of the namespaces.
+
+Lets create the veth interface:
+```
+sudo ip link add veth00 type veth peer name veth10
+```
+And place each in the relevant namespace:
+```
+sudo ip link set dev veth00 netns red
+sudo ip link set dev veth10 netns blue
+```
+
+At this point, assuming that IPv6 is enabled on the host, each namespace
+should show an loopback interface and another veth type interface which
+has a [link-local IPv6 address](https://tools.ietf.org/html/rfc4291#page-11).
+
+To check the interfaces and their addresses, execute:
+```
+sudo ip netns exec red ip addr
+sudo ip netns exec blue ip addr
+```
+
+## L3 connectivity
+With the IPv6 link-local addresses in place, we can already check the
+connectivity betweem the two namespaces.
+
+Using the infromation gathered in the previous `ip addr` commands, learn
+the link-local IPv6 address of each peer and use it in the ping command.
+
+Note: As the addresses have link-local scope, a zone must be added to the
+destination address.
+A link-local address has a default subnet of 64 bits with a default network
+address of `fe80`. Therefore, on a node with multiple interfaces, an explicit
+egress interface needs to be provided in order for the packet to know through
+which interface to exit. 
+See [here](https://tools.ietf.org/html/rfc4007) for more information
+about zones.
+
+Lets run an IPv6 ping:
+(on older distributions use `ping -6` and on newer ones use `ping6`)
```

Comment:
According to my understanding, `ping6` is the old one, and on newer distros you can just use `ping`. `ping -6` is required, if you explicitly wants to enforce the usage of IPv6 because the destination might be resolved  to IPv4, which is not possible if you explicitly specify an IPv6 address as destination.

Reply:
Right, I am removing this line.